### ISSUE TITLE
float <-> int, int64 -> int, int -> char conversions

### ISSUE THREAD

**Issue by suamor** [canorusmusic/canorus#122]
There are a large number of warnings regarding float <-> int, int64 -> int, int -> char conversions that do not use static_cast.
Here is a list with files having these issues:
- layoutengine.cpp
- archive.cpp
- muselementfactory.cpp
- musicxmlexport.cpp
- canorusmlimport.cpp
- drawablenote.cpp

They are not warned by compiler but by qt-lint alike processing of the qtcreator editor.

Those should be checked for boundaries / valid conversions (also comments would help better understanding why this is necessary).

List will be updated should I find more of those issues

**Comment by matevz** [comment]
Note that I fixed some of those along in #99, so I suggest you rebase asap.

**Comment by suamor** [comment]
Yes, that is my plan as soon as I'm through with the code. I want to finish first though because most likely merging will be pita...

**Comment by suamor** [comment]
I reviewed and did a > 50% of warning removal on scoreview.cpp and scoreview.h.
I added many notes in that files. Still to not forget it a summary of my concerns:
- Coordinate conversion from int <-> double (float is also still lingering) -> could have precision and performance issues (mainly for coordinates but also for time conversion)
- a header inline seems to be broken (partially ?)
- parts where I fixed "simple" cast warnings should be reviewed

**Comment by matevz** [comment]
Please open PR for this asap, so I can make a review.

**Comment by suamor** [comment]
https://github.com/canorusmusic/canorus/pull/123/

**Comment by matevz** [comment]
Was this fixed by #123?
